internal/kafka/command_topic.go

```diff
@@ -206,13 +206,23 @@ func addApiKeyToCluster(cmd *cobra.Command, cluster *config.KafkaClusterConfig)
 	return nil
 }
 
-func ProduceToTopic(cmd *cobra.Command, keyMetaInfo []byte, valueMetaInfo []byte, topic string, keySerializer serdes.SerializationProvider, valueSerializer serdes.SerializationProvider, producer *ckgo.Producer) error {
+func (c *command) produceToTopic(cmd *cobra.Command, keyMetaInfo []byte, valueMetaInfo []byte, topic string, keySerializer serdes.SerializationProvider, valueSerializer serdes.SerializationProvider, producer *ckgo.Producer, isOnPrem bool) error {
 	keys := "Ctrl-C or Ctrl-D"
 	if runtime.GOOS == "windows" {
 		keys = "Ctrl-C"
 	}
 	output.ErrPrintf(false, "Starting Kafka Producer. Use %s to exit.\n", keys)
-
+	if isOnPrem {
+		go func(eventsChan chan ckgo.Event) {
+			for ev := range eventsChan {
+				oart, _ := ev.(ckgo.OAuthBearerTokenRefresh)
+				err := c.refreshOAuthBearerToken(cmd, producer, oart)
+				if err != nil {
+					return
+				}
```

Comment:
I saw on a previous commit that the `, ok :=` was removed. Could we add it back, but before `c.refreshOAuthBearerToken`, or does that introduces issues?